temporalio/activity.py

```diff
@@ -352,7 +353,11 @@ def from_callable(fn: Callable) -> Optional[_Definition]:
     def must_from_callable(fn: Callable) -> _Definition:
         ret = _Definition.from_callable(fn)
         if ret:
-            return ret
+            # We have to replace the function with the given callable here
+            # because the one passed in may be a method or some other partial
+            # that represents the real callable instead of what the decorator
+            # used.
+            return dataclasses.replace(ret, fn=fn)
```

Comment:
I'm probably missing context but why here and not in `from_callable` or even when generating `__temporal_activity_definition`?

Reply:
> I'm probably missing context but why here and not in `from_callable`

You're not missing any context, I should have put this there. Will change on next commit (or on its own assuming all else is good).

> when generating `__temporal_activity_definition`

This is what I do, but the problem is for activity methods (which we should support), the decorator is invoked on the unbound method but the callable coming in for `from_callable` is at call time has `self` bound. I want to have the bound method as `fn`.

Reply:
Fixed